Library/Homebrew/extend/os/mac/requirements.rb

```diff
@@ -0,0 +1,5 @@
+class SubversionRequirementMac < Requirement
+  satisfy { true }
```

Comment:
Maybe `which "subversion"` which is equivalent?

Reply:
Also: thoughts on doing the opposite and making just a Linux requirement with the same name `SubversionRequirement` and just defines `default_formula`? Feel free to disagree; I realise you may have tried that.

Reply:
Scratch the last comment: I no longer like it. Interested why this uses a different name, though.

Reply:
If it uses the same name `class SubversionRequirement`, the behaviour is to re-open and modify the existing generic class. That'd need something like `deafault_formula nil` to override the `default_formula` specified by the generic class. `satisfy { true }` also fails to replace existing `satisfied?` method due to this caching:
```ruby
@satisfied ||= Requirement::Satisfier.new(options, &block)
```
https://github.com/Homebrew/brew/blob/master/Library/Homebrew/requirement.rb#L174

It seemed cleaner to just replace the existing generic class, rather than extend it.

Reply:
Ok, seems fine. Let's just use `which "subversion"` here and I think it's good to go.

Reply:
Won't that change the current behavior and use whatever subversion is in the user's PATH?

Reply:
Also: isn't this supposed to be the purview of the svn superenv shim?

Reply:
Actually, not quite: it seems the version check below should still work on macOS if you need to require a version that's higher than the system version. @ilovezfs, thoughts?

Reply:
> Actually, not quite:

Reply:
> Actually, not quite: 

Posted before @ilovezfs's comment showed up. Yes, good points about the `svn` superenv shim. @sjackman does that not work right on Linux? What's the difference between Subversion and Git, here?

Reply:
If we see such a situation in the wild where macOS system version is insufficient, we could consider revisiting this, but I don't think that is likely. And without a default formula, using `which "svn"` (I assume you meant `which "svn"` not `which "subversion"`) is a security vulnerability because it won't ensure the installed version of the formula is up to date if that's the svn first in the PATH.

Reply:
> is a security vulnerability because it won't ensure the installed version of the formula is up to date if that's the svn first in the PATH.

It will, actually; if a requirement's executable resolves to a formula then it'll upgrade it.

Reply:
OK. I still think this is a bad idea because it allows the dependencies of subversion to be involved in the macOS builds unnecessarily.

Reply:
Also Homebrew svn doesn't recover from a canceled build.
```
==> Checking out 3094
svn up /Users/joe/Library/Caches/Homebrew/netpbm--svn -r 3094
svn: E155004: Run 'svn cleanup' to remove locks (type 'svn help cleanup' for details)
svn: E155004: Working copy '/Users/joe/Library/Caches/Homebrew/netpbm--svn' locked.
svn: E155004: '/Users/joe/Library/Caches/Homebrew/netpbm--svn' is already locked.
Error: Failed to download resource "netpbm"
Failure while executing: svn up /Users/joe/Library/Caches/Homebrew/netpbm--svn -r 3094
```

Reply:
It also fails to respect `export HOMEBREW_SVN=/usr/bin/svn` which simply hangs.

Reply:
`which "subversion"` will also make `brew pin subversion` impossible.

Reply:
I'm happy with either `true` or `which "svn"`. Let me know your preference or whether you need any additional input from me.

Reply:
@sjackman I guess I'm interested on why a `SubversionRequirement` is needed but `GitRequirement` is not? FWIW if this PR (or a new PR) is scoped to just the `CVSRequirement` I'm happy to merge that as-is.

Reply:
> I guess I'm interested on why a SubversionRequirement is needed but GitRequirement is not?

To expand on that further: both provide shims in `Library/Homebrew/shims/scm/` so it would be good to use a similar method to `Utils.git_path` in any `Requirement` if we need one (which the lack of a `GitRequirement` draws into question)

Reply:
`GitRequirement` does exist. Does that answer your question, or did I misunderstand?
https://github.com/Homebrew/brew/blob/master/Library/Homebrew/requirements.rb#L124
https://github.com/Homebrew/brew/blob/master/Library/Homebrew/dependency_collector.rb#L153-L154

Reply:
Why wouldn't an svn requirement do the same shim-aware
```
  def self.git_available?
    @git ||= quiet_system HOMEBREW_SHIMS_PATH/"scm/git", "--version"
  end
```

Reply:
and the same automatic install thing https://github.com/Homebrew/brew/blob/82aa148cb262cad831daa800da4f02d1b6e95dbf/Library/Homebrew/utils/git.rb#L47-L61

Reply:
I think it's fine to have the automatic install be through the `default_formula` like Git is but I agree a `satisfy { Utils.svn_available? }` would be a good idea. Sorry for all the back and forth here, @sjackman.

Reply:
I think you meant for macOS, so I've changed `satisfy { true }` to `satisfy { Utils.svn_available? }`.

Reply:
@sjackman Sorry for all the run around here but I think we can have a single, generic `SubversionRequirement` with a `default_formula` after all that checks `Utils.svn_available?` (i.e. exactly mirrors the `GitRequirement`). A version would be a nice addition if the tag is specified but I think you could punt on that for now if you wanted, too.

Apologies again for this; my knowledge of this part of the codebase is a bit rusty.

Reply:
:+1:

Reply:
Done!